content/en/blog/deploy_kmesh_in_asm/index.md

```diff
@@ -0,0 +1,530 @@
+
+### Introduction to Kmesh 
+Kmesh's dual-engine mode uses eBPF to intercept traffic in kernel space and deploys a Waypoint Proxy to handle complex L7 traffic management, thus separating L4 and L7 governance between kernel space (eBPF) and user space (Waypoint). Compared to Istio's Ambient Mesh, it reduces latency by 30%. Compared to the kernel-native mode, the dual-engine mode does not require kernel enhancements, offering broader applicability.
+
+![Dual-Engine Mode](images/dual-engine-mode.png)
+
+Currently, ASM supports using Kmesh's dual-engine mode as one of the data planes for the service mesh, enabling more efficient service management. Specifically, ASM can be used as the control plane, while Kmesh can be deployed as the data plane within an Alibaba Cloud Container Service for Kubernetes (ACK) cluster.
+
+## Deploy Kmesh in ACK and Connect to ASM
+### Prerequisites
+Create an ASM cluster and add the ACK cluster to the ASM cluster for management. For detailed steps, you can refer to the documentation: [Add a cluster to an ASM instance](https://www.alibabacloud.com/help/en/asm/getting-started/add-a-cluster-to-an-asm-instance-1?spm=a2c63.l28256.help-menu-search-147365.d_0).
+
+### Install Kmesh
+Run the following command to clone the Kmesh project into your local machine.
+
+```shell
+git clone https://github.com/kmesh-net/kmesh.git && cd kmesh
+```
+
+#### Check Services of ASM Control Plane
+After the Kmesh is downloaded, you need to execute the following command first to check the Service name of the current ASM control plane in the cluster, in order to configure the connection between Kmesh and the ASM control plane.
+
+```shell
+kubectl get svc -n istio-system | grep istiod
+
+# istiod-1-22-6   ClusterIP   None   <none>   15012/TCP   2d
+```
+
+#### Install Kmesh with Kubectl
+You can use kubectl to install Kmesh in the ACK Kubernetes cluster. However, before installation, please add the `ClusterId` and `xdsAddress` environment variables to the Kmesh DaemonSet. These are used for the authentication and connection between Kmesh and the ASM control plane. The ClusterId is the ID of the ACK cluster where Kmesh is deployed, and the xdsAddress is the Service of the ASM control plane.
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kmesh
+  labels:
+    app: kmesh
+  namespace: kmesh-system
+spec:
+    spec:
+      containers:
+        - env:
+          # ASM Control Plane Service
+          - name: XDS_ADDRESS
+            value: "istiod-1-22-6.istio-system.svc:15012"
+          # add ACK cluster id
+          - name: CLUSTER_ID
+            value: "cluster-id"
+    ...
+```
+
+After modifying the Kmesh DaemonSet environment variables in the following command, execute it to deploy Kmesh.
+
+```shell
```

Comment:
Can you use helm install?

Reply:
Because we have some restriction about creating envoyfilter for compatibility, to be more specific, fields like `proxyVersion` and `labels` are required or the creation will be denied by the webhook. So when I write the document I put a complete yaml here. But I think you are right, I should just keep the part which describes configuration of `XDS_ADDRESS` and `CLUSTER_ID` environments and remove the long yaml, that is what we should focus on.

Reply:
By the way, it seems like `CLUSTER_ID` environment can not be set via helm values, maybe I can do that?

Reply:
We need to provide such option later?

Reply:
I think if it is provided would be better. :)